### PR TITLE
Use Jax RNG for data shuffling in MNIST example.

### PR DESCRIPTION
**Pull request by danielsuo**
This PR contains commit history from #293 and fixes up `howto`s.

### CODE REVIEW

File: howtos/diffs/checkpointing.diff [hunk 1]
```diff
@@ -11,23 +11,23 @@ index 5dc74db..975e50f 100644
...
+@@ -198,6 +202,10 @@ def train(train_ds, test_ds):
      summary_writer.scalar('train_accuracy', train_metrics['accuracy'], epoch)
      summary_writer.scalar('eval_loss', loss, epoch)
      summary_writer.scalar('eval_accuracy', accuracy, epoch)
-+    
-+    # Save model at every epoch
-+    checkpoints.save_checkpoint(FLAGS.model_dir, optimizer, epoch, keep=3)
-+  
++
++  # Save model at every epoch
```
Comment (avital) [inline howtos/diffs/checkpointing.diff:28]: Is this correct? The indentation look to have changed -- doesn't this now only save the model at the end of training?

File: howtos/diffs/distributed-training.diff [hunk 2]
```diff
@@ -62,21 +62,21 @@ index 5dc74db..77b49fa 100644
...
-@@ -192,7 +207,8 @@ def train(train_ds, test_ds):
-   for epoch in range(1, num_epochs + 1):
+@@ -191,7 +206,8 @@ def train(train_ds, test_ds):
+     rng, input_rng = random.split(rng)
      optimizer, train_metrics = train_epoch(
          optimizer, train_ds, batch_size, epoch, input_rng)
 -    loss, accuracy = eval_model(optimizer.target, test_ds)
 +    model = jax_utils.unreplicate(optimizer.target)  # Fetch from 1st device
-+    loss, accuracy = eval_model(model, test_ds)
++    loss_accuracy = eval_model(model, test_ds)
```
Comment (avital) [inline howtos/diffs/distributed-training.diff:79]: Is this correct? Seem like a typo maybe? (Do HOWTO tests pass here? -- looks like yes, but maybe the tests don't yet cover this step?)

Comment (danielsuo) [reply howtos/diffs/distributed-training.diff:79]: Yes, this is a typo! The test tests `train_step`, but not `train`.

File: howtos/diffs/checkpointing.diff (hunk 1 above)
Comment (danielsuo) [reply howtos/diffs/checkpointing.diff]: Yes, that sounds right; fixed.
